skrub/_reporting/tests/test_table_report.py

```diff
@@ -123,6 +126,49 @@ def test_duration(df_module):
     assert re.search(r"2(\.0)?\s+days", TableReport(df).html())
 
 
+@pytest.mark.parametrize(
+    "filename_type",
+    ["str", "Path", "file_object", "binary_mode"],
+)
+def test_write_html(tmp_path, pd_module, filename_type):
+    df = pd_module.make_dataframe({"a": [1, 2], "b": [3, 4]})
+    report = TableReport(df)
+
+    tmp_file_path = tmp_path / Path("report.html")
+
+    if filename_type == "str":
+        filename = str(tmp_file_path)
+    elif filename_type == "file_object":
+        filename = open(tmp_file_path, "w", encoding="utf-8")
+    elif filename_type == "binary_mode":
+        filename = open(tmp_file_path, "wb")
+    else:
+        filename = tmp_file_path
+
+    report.write_html(filename)
+    assert tmp_file_path.exists()
+
+    with open(tmp_file_path, "r", encoding="utf-8") as file:
+        saved_content = file.read()
+    assert "</html>" in saved_content
+
+
+def test_write_html_with_not_utf8_encoding(tmp_path, pd_module):
+    df = pd_module.make_dataframe({"a": [1, 2], "b": [3, 4]})
+    report = TableReport(df)
+
+    filename = open(tmp_path / Path("report.html"), "w", encoding="latin-1")
+    encoding = getattr(filename, "encoding", None)
+    with pytest.raises(
+        ValueError,
+        match=(
+            f"If `file` is a text file it should use utf-8 encoding; got: {encoding!r}"
+        ),
+    ):
+        report.write_html(filename)
+        assert not filename.exists()
```

Comment:
```suggestion
        report.write_html(filename)
    assert not filename.exists()
```

otherwise the assert is never executed because of the exception raised on the line above

Reply:
I actually change this to check if the file doesn't contain `html` to make sure we don't modify it.